airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ConnectionName.tsx

```diff
@@ -0,0 +1,151 @@
+import { faPenToSquare } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ChangeEvent, useState } from "react";
+import styled from "styled-components";
+
+import { Input } from "components";
+
+import { WebBackendConnectionRead } from "core/request/AirbyteClient";
+import { useUpdateConnection } from "hooks/services/useConnectionHook";
+
+type Props = {
+  connection: WebBackendConnectionRead;
+};
+
+const MainContainer = styled.div`
+  margin-top: 10px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const Icon = styled(FontAwesomeIcon)`
+  display: none;
+  position: absolute;
+  right: 20px;
+  font-size: 18px;
+  color: ${({ theme }) => theme.primaryColor};
+`;
+
+const NameContainer = styled.div`
+  width: 650px;
+  background-color: rgba(255, 235, 215, 0.4);
+  display: flex;
+  align-items: center;
+  position: relative;
+  padding: 0 20px;
+  border-radius: 8px;
+  border: 1px solid rgba(255, 235, 215, 0.4);
+
+  &:hover {
+    cursor: pointer;
+    border: ${({ theme }) => `1px solid ${theme.primaryColor}`};
+    background-color: ${({ theme }) => theme.primaryColor12};
+  }
+
+  &:hover ${Icon} {
+    display: block;
+  }
+`;
+
+const EditingContainer = styled.div`
+  width: 650px;
+  display: flex;
+  background-color: white;
+  justify-content: center;
+  align-items: center;
+  border-radius: 8px;
+  border: ${({ theme }) => `1px solid ${theme.primaryColor}`};
+`;
+
+const InputContainer = styled.div`
+  height: 50px;
+  width: 100%;
+
+  div {
+    border-radius: 8px;
+    border: none;
+    box-shadow: none;
+    background-color: white !important;
+  }
+`;
+
+const Name = styled.div`
+  flex-grow: 1;
+`;
+
+const H2 = styled.h2`
+  font-weight: 700;
+  font-size: 24px;
+  line-height: 29px;
+  text-align: center;
+  color: #1a194d;
+  margin: 10px;
+`;
+
+const StyledInput = styled(Input)`
+  border-radius: 8px;
+  background-color: white;
+  font-size: 24px;
+  height: 50px;
+  div {
+    border: none;
+  }
+`;
+
+const ConnectionName: React.FC<Props> = ({ connection }) => {
+  const { name } = connection;
+  const [editingState, setEditingState] = useState(false);
+  const [connectionName, setConnectionName] = useState(connection.name);
+  const { mutateAsync: updateConnection } = useUpdateConnection();
+
+  const setEditing = () => {
+    setEditingState(true);
+  };
+
+  const inputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+    if (value) {
+      setConnectionName(event.currentTarget.value);
+    }
+  };
+
+  const onBlur = async () => {
+    const mustPassParameters = {
+      connectionId: connection.connectionId,
+      namespaceDefinition: connection.namespaceDefinition,
+      syncCatalog: connection.syncCatalog,
+      status: connection.status,
+      prefix: connection.prefix,
+    };
+
+    await updateConnection({
```

Comment:
@timroes Here's a situation that concerns me in that, to update the connection name, we need to dump the entire state of the connection because we cannot request a partial update. If the connection model changes in the future we also have to update this code. I think this is ok as a temporary measure and in an effort to get the work on the PR complete, but it may make sense to request an api call that only updates the connection name.

Reply:
Yeah this will be helpful either way
1. A new API call
2. The API takes only which needs to be updated